ext/rbs_extension/unescape.c

```diff
@@ -19,6 +19,7 @@ void rbs_unescape_string(VALUE string) {
 
   if (!HASH) {
     HASH = rb_hash_new();
+    rb_global_variable(&HASH);
```

Comment:
GC may be started from inside `rb_str_new_literal` call where the `HASH` is not placed on the stack. It means the value of `HASH` may be freed before the execution of `rb_global_variable` at the lsat place.

So, moving the `rb_global_variable` here will ensure it is marked before any possible GC start. 💪